// merchi_checkout/lib/components/buttons/ButtonNext.js
'use client';

import { useMerchiCheckboutContext } from '../MerchiCheckoutProvider';
import { CgSpinner } from 'react-icons/cg';
import { FaArrowRight } from 'react-icons/fa';
import { jsx as _jsx } from "react/jsx-runtime";
function ButtonNext(_ref) {
  var formId = _ref.formId,
    forceDisabled = _ref.forceDisabled,
    loading = _ref.loading,
    onClick = _ref.onClick;
  var _useMerchiCheckboutCo = useMerchiCheckboutContext(),
    activeTabIndex = _useMerchiCheckboutCo.activeTabIndex,
    classNameMerchiCheckoutButtonPrimary = _useMerchiCheckboutCo.classNameMerchiCheckoutButtonPrimary,
    next = _useMerchiCheckboutCo.nextTab,
    tabs = _useMerchiCheckboutCo.tabs;
  var nextTab = tabs[(activeTabIndex || 0) + 1];
  var handleButtonClick = onClick ? onClick : next;
  var disabled = nextTab && nextTab.disabled;
  if (![null, undefined].includes(forceDisabled)) {
    disabled = forceDisabled;
  }
  return /*#__PURE__*/_jsx("button", {
    type: formId ? 'submit' : 'button' // Set type to "submit" if formId is provided
    ,
    form: formId // Associate this button with the form using formId
    ,
    className: classNameMerchiCheckoutButtonPrimary + ' btn-lg',
    disabled: disabled || loading,
    onClick: !formId ? handleButtonClick : undefined,
    children: loading ? /*#__PURE__*/_jsx(CgSpinner, {
      fontSize: "1.1rem",
      className: "animate_spin"
    }) : /*#__PURE__*/_jsx(FaArrowRight, {
      fontSize: "1.1rem"
    })
  });
}
export default ButtonNext;
