// merchi_checkout/lib/components/forms/InputAcceptUserTermsAndConditions.js
'use client';

import { useMerchiCheckboutContext } from '../MerchiCheckoutProvider';
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
function InputAcceptUserTermsAndConditions(_ref) {
  var isChecked = _ref.isChecked,
    setIsChecked = _ref.setIsChecked;
  var _useMerchiCheckboutCo = useMerchiCheckboutContext(),
    classNameMerchiCheckoutFormCheckbox = _useMerchiCheckboutCo.classNameMerchiCheckoutFormCheckbox,
    classNameMerchiCheckoutFormGroupCheckbox = _useMerchiCheckboutCo.classNameMerchiCheckoutFormGroupCheckbox,
    classNameMerchiCheckoutFormLabelCheckbox = _useMerchiCheckboutCo.classNameMerchiCheckoutFormLabelCheckbox;
  return /*#__PURE__*/_jsx("div", {
    onClick: function onClick() {
      return setIsChecked(!isChecked);
    },
    className: classNameMerchiCheckoutFormGroupCheckbox,
    children: /*#__PURE__*/_jsxs("label", {
      className: classNameMerchiCheckoutFormLabelCheckbox,
      children: [/*#__PURE__*/_jsx("input", {
        className: classNameMerchiCheckoutFormCheckbox,
        defaultChecked: isChecked,
        type: "checkbox"
      }), ' ', /*#__PURE__*/_jsxs("span", {
        style: {
          fontSize: '14px'
        },
        className: "text-muted",
        children: ["I agree to the user profile", ' ', /*#__PURE__*/_jsx("a", {
          href: "/terms-and-conditions/user/",
          target: "_blank",
          children: "terms & conditions"
        }), "."]
      })]
    })
  });
}
export default InputAcceptUserTermsAndConditions;
