// merchi_checkout/lib/components/forms/InputSelect.js
'use client';

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
import { v4 as uuidv4 } from 'uuid';
import { useController } from 'react-hook-form';
import { useMerchiCheckboutContext } from '../MerchiCheckoutProvider';
import InputError from './InputError';
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
function InputSelect(_ref) {
  var control = _ref.control,
    label = _ref.label,
    name = _ref.name,
    options = _ref.options,
    rules = _ref.rules;
  var _useMerchiCheckboutCo = useMerchiCheckboutContext(),
    classNameMerchiCheckoutFormGroup = _useMerchiCheckboutCo.classNameMerchiCheckoutFormGroup,
    classNameMerchiCheckoutFormInput = _useMerchiCheckboutCo.classNameMerchiCheckoutFormInput;
  var _useController = useController({
      name: name,
      control: control,
      rules: rules,
      defaultValue: ''
    }),
    field = _useController.field,
    _useController$fieldS = _useController.fieldState,
    error = _useController$fieldS.error,
    invalid = _useController$fieldS.invalid;
  return /*#__PURE__*/_jsxs("div", {
    className: classNameMerchiCheckoutFormGroup,
    children: [label && /*#__PURE__*/_jsx("label", {
      children: label
    }), /*#__PURE__*/_jsx("select", _objectSpread(_objectSpread({}, field), {}, {
      className: "".concat(classNameMerchiCheckoutFormInput, " ").concat(invalid && ' is-invalid'),
      children: options.map(function (option, index) {
        return /*#__PURE__*/_jsx("option", {
          value: option.value,
          children: option.label
        }, index + uuidv4());
      })
    })), /*#__PURE__*/_jsx(InputError, {
      error: error
    })]
  });
}
export default InputSelect;
