// merchi_checkout/lib/components/Alerts.js
'use client';

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
import { useEffect } from 'react';
import { useMerchiCheckboutContext } from './MerchiCheckoutProvider';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCheckCircle, faExclamationCircle, faExclamationTriangle, faInfoCircle, faTimes } from '@fortawesome/free-solid-svg-icons';
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
var iconsMap = {
  danger: faExclamationCircle,
  info: faInfoCircle,
  success: faCheckCircle,
  warning: faExclamationTriangle
};
function AlertSmall(_ref) {
  var _ref$alertType = _ref.alertType,
    alertType = _ref$alertType === void 0 ? 'error' : _ref$alertType,
    close = _ref.close,
    _ref$delay = _ref.delay,
    delay = _ref$delay === void 0 ? 5000 : _ref$delay,
    message = _ref.message,
    title = _ref.title;
  var _useMerchiCheckboutCo = useMerchiCheckboutContext(),
    classNameMerchiCheckoutAlert = _useMerchiCheckboutCo.classNameMerchiCheckoutAlert,
    classNameMerchiCheckoutAlertError = _useMerchiCheckboutCo.classNameMerchiCheckoutAlertError,
    classNameMerchiCheckoutAlertSuccess = _useMerchiCheckboutCo.classNameMerchiCheckoutAlertSuccess;
  function setTimer() {
    setTimeout(close, delay);
  }
  useEffect(function () {
    if (!['danger', 'error'].includes(alertType)) {
      setTimer();
    }
  }, [message]);
  var classNameAlertType = alertType === 'error' ? classNameMerchiCheckoutAlertError : classNameMerchiCheckoutAlertSuccess;
  return /*#__PURE__*/_jsx("div", {
    className: "".concat(classNameMerchiCheckoutAlert, " ").concat(classNameAlertType),
    children: /*#__PURE__*/_jsxs("div", {
      style: {
        display: 'flex',
        justifyContent: 'space-between'
      },
      children: [/*#__PURE__*/_jsx(FontAwesomeIcon, {
        className: "alert-icon",
        icon: iconsMap[alertType]
      }), /*#__PURE__*/_jsxs("div", {
        className: "alert-text",
        children: [/*#__PURE__*/_jsx("span", {
          className: "alert-title",
          "data-notify": "title",
          children: title
        }), ' ', /*#__PURE__*/_jsx("span", {
          "data-notify": "message",
          children: message
        })]
      }), /*#__PURE__*/_jsx("div", {
        onClick: close,
        children: /*#__PURE__*/_jsx(FontAwesomeIcon, {
          icon: faTimes
        })
      })]
    })
  });
}
function Alerts() {
  var _useMerchiCheckboutCo2 = useMerchiCheckboutContext(),
    alerts = _useMerchiCheckboutCo2.alerts,
    alertClose = _useMerchiCheckboutCo2.alertClose,
    classNameMerchiCheckoutAlertsContainer = _useMerchiCheckboutCo2.classNameMerchiCheckoutAlertsContainer;
  ;
  return /*#__PURE__*/_jsx("div", {
    className: classNameMerchiCheckoutAlertsContainer,
    children: alerts.map(function (alert, index) {
      return /*#__PURE__*/_jsx(AlertSmall, _objectSpread({
        close: function close() {
          return alertClose(index);
        }
      }, alert), "alert-index-key-".concat(index));
    })
  });
}
export default Alerts;
