// merchi_checkout/lib/components/DraftImageUploaded.js
'use client';

import { faTrashAlt } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { useMerchiCheckboutContext } from './MerchiCheckoutProvider';
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
function isPdf(file) {
  return ['application/pdf', 'application/x-pdf'].includes(file.mimetype);
}
function DraftImageUploaded(_ref) {
  var doDelete = _ref.doDelete,
    image = _ref.image;
  var _useMerchiCheckboutCo = useMerchiCheckboutContext(),
    classNameMerchiCheckoutButtonPrimary = _useMerchiCheckboutCo.classNameMerchiCheckoutButtonPrimary;
  return /*#__PURE__*/_jsxs("div", {
    className: "upload-draft-image-container",
    children: [/*#__PURE__*/_jsx("div", {
      className: "text-center",
      children: isPdf(image) ? /*#__PURE__*/_jsx("i", {
        className: "far fa-file-pdf"
      }) : /*#__PURE__*/_jsx("div", {
        className: "image-preview-div",
        style: {
          backgroundImage: "url(".concat(image.viewUrl, ")")
        }
      })
    }), doDelete && /*#__PURE__*/_jsx("button", {
      className: classNameMerchiCheckoutButtonPrimary,
      onClick: doDelete,
      children: /*#__PURE__*/_jsx(FontAwesomeIcon, {
        icon: faTrashAlt
      })
    })]
  });
}
export default DraftImageUploaded;
export function DraftImagesStatic(_ref2) {
  var items = _ref2.items,
    _ref2$className = _ref2.className,
    className = _ref2$className === void 0 ? 'width-150 z-index-10000' : _ref2$className;
  return /*#__PURE__*/_jsx("div", {
    className: "d-flex justify-content-center flex-wrap z-index-10000",
    children: items.map(function (image, i) {
      return /*#__PURE__*/_jsx("div", {
        className: className,
        children: /*#__PURE__*/_jsx(DraftImageUploaded, {
          image: image
        })
      }, "".concat(i, "-draft-image"));
    })
  });
}
