// merchi_checkout/lib/components/MerchiCheckoutProvider.js
'use client';

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _toConsumableArray(arr) { return _arrayWithoutHoles(arr) || _iterableToArray(arr) || _unsupportedIterableToArray(arr) || _nonIterableSpread(); }
function _nonIterableSpread() { throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _iterableToArray(iter) { if (typeof Symbol !== "undefined" && iter[Symbol.iterator] != null || iter["@@iterator"] != null) return Array.from(iter); }
function _arrayWithoutHoles(arr) { if (Array.isArray(arr)) return _arrayLikeToArray(arr); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t["return"] && (u = t["return"](), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import * as React from 'react';
import { createContext, useContext, useEffect, useState } from 'react';
import { tabsInit } from '../tabs_utils';
import { jsx as _jsx } from "react/jsx-runtime";
var MerchiCheckoutContext = /*#__PURE__*/createContext({
  activeTabIndex: 0,
  alerts: [],
  alertClose: function alertClose() {},
  alertErrorShow: function alertErrorShow() {},
  classNameMerchiCheckoutAlertsContainer: undefined,
  classNameMerchiCheckoutAlert: undefined,
  classNameMerchiCheckoutAlertError: undefined,
  classNameMerchiCheckoutAlertSuccess: undefined,
  classNameMerchiCheckoutButtonPrimary: undefined,
  classNameMerchiCheckoutButtonPrimaryBlock: undefined,
  classNameMerchiCheckoutButtonSecondary: undefined,
  classNameMerchiCheckoutButtonSecondaryBlock: undefined,
  classNameMerchiCheckoutButtonDownloadInvoice: undefined,
  classNameMerchiCheckoutFormCheckbox: undefined,
  classNameMerchiCheckoutFormGroup: undefined,
  classNameMerchiCheckoutFormGroupCheckbox: undefined,
  classNameMerchiCheckoutFormInput: undefined,
  classNameMerchiCheckoutGoogleSuggestList: undefined,
  classNameMerchiCheckoutGoogleSuggestListItem: undefined,
  classNameMerchiCheckoutListGroupItemLoader: undefined,
  classNameMerchiCheckoutInputError: undefined,
  classNameMerchiCheckoutListGroup: undefined,
  classNameMerchiCheckoutListGroupItem: undefined,
  classNameMerchiCheckoutRow: undefined,
  classNameMerchiCheckoutRowColumn: undefined,
  classNameMerchiCheckoutTabsContainer: undefined,
  classNameMerchiCheckoutFormLabelCheckbox: undefined,
  classNameMerchiCheckoutTab: undefined,
  classNameMerchiCheckoutTabPane: undefined,
  classNameMerchiCheckoutTabPaneContainer: undefined,
  classNameMerchiCheckoutTabButton: undefined,
  classNameMerchiInvoiceButtonPayInvoice: undefined,
  clearCustomer: function clearCustomer() {},
  currentUser: undefined,
  customer: undefined,
  discountButtonText: undefined,
  discountCallbackError: function discountCallbackError() {},
  discountCallbackSuccess: function discountCallbackSuccess() {},
  discountClassName: undefined,
  discountClassNameButton: undefined,
  discountClassNameButtonContainer: undefined,
  discountClassNameButtonItemRemove: undefined,
  discountClassNameErrorMessage: undefined,
  discountClassNameInput: undefined,
  discountClassNameInputContainer: undefined,
  discountClassNameInputdiscountLabel: undefined,
  discountClassNameListItem: undefined,
  discountClassNameListItems: undefined,
  discountClassNameMainContainer: undefined,
  discountLabel: undefined,
  discountShowAppliedItems: true,
  domain: undefined,
  editDraftTemplate: function editDraftTemplate() {},
  googlePlacesApiKey: undefined,
  googlePlacesLoaded: false,
  isBuyRequest: false,
  isOpen: false,
  loading: false,
  messageSuccessBuyRequest: undefined,
  messageSuccessQuoteRequest: undefined,
  nextTab: function nextTab() {},
  includeDomainSignup: false,
  invoice: {},
  isProductEmbedForm: false,
  job: {},
  product: {},
  redirectAfterSuccessUrl: undefined,
  redirectAfterQuoteSuccessUrl: undefined,
  redirectWithValue: true,
  setActiveTabById: function setActiveTabById() {},
  setActiveTabIndex: function setActiveTabIndex() {},
  setCustomer: function setCustomer() {},
  setDomain: function setDomain() {},
  setInvoice: function setInvoice() {},
  setJob: function setJob() {},
  showDiscountCode: true,
  showUserTermsAndConditions: true,
  tabs: [],
  toggleMerchiCheckout: function toggleMerchiCheckout() {},
  urlFrontend: '',
  urlApi: ''
});
export var useMerchiCheckboutContext = function useMerchiCheckboutContext() {
  return useContext(MerchiCheckoutContext);
};
export var MerchiCheckoutProvider = function MerchiCheckoutProvider(_ref) {
  var children = _ref.children,
    _ref$classNameMerchiC = _ref.classNameMerchiCheckoutAlertsContainer,
    classNameMerchiCheckoutAlertsContainer = _ref$classNameMerchiC === void 0 ? 'd-flex justify-content-center flex-column alerts-container-sm' : _ref$classNameMerchiC,
    _ref$classNameMerchiC2 = _ref.classNameMerchiCheckoutAlert,
    classNameMerchiCheckoutAlert = _ref$classNameMerchiC2 === void 0 ? 'alert alert-dismissible alert-notify' : _ref$classNameMerchiC2,
    _ref$classNameMerchiC3 = _ref.classNameMerchiCheckoutAlertError,
    classNameMerchiCheckoutAlertError = _ref$classNameMerchiC3 === void 0 ? 'alert-danger' : _ref$classNameMerchiC3,
    _ref$classNameMerchiC4 = _ref.classNameMerchiCheckoutAlertSuccess,
    classNameMerchiCheckoutAlertSuccess = _ref$classNameMerchiC4 === void 0 ? 'alert-success' : _ref$classNameMerchiC4,
    _ref$classNameMerchiC5 = _ref.classNameMerchiCheckoutButtonPrimary,
    classNameMerchiCheckoutButtonPrimary = _ref$classNameMerchiC5 === void 0 ? 'btn btn-lmd btn-primary d-flex align-items-center justify-content-center' : _ref$classNameMerchiC5,
    _ref$classNameMerchiC6 = _ref.classNameMerchiCheckoutButtonPrimaryBlock,
    classNameMerchiCheckoutButtonPrimaryBlock = _ref$classNameMerchiC6 === void 0 ? 'btn btn-lg btn-primary btn-block' : _ref$classNameMerchiC6,
    _ref$classNameMerchiC7 = _ref.classNameMerchiCheckoutButtonSecondary,
    classNameMerchiCheckoutButtonSecondary = _ref$classNameMerchiC7 === void 0 ? 'btn btn-lg btn-secondary' : _ref$classNameMerchiC7,
    _ref$classNameMerchiC8 = _ref.classNameMerchiCheckoutButtonSecondaryBlock,
    classNameMerchiCheckoutButtonSecondaryBlock = _ref$classNameMerchiC8 === void 0 ? 'btn btn-lg btn-secondary btn-block' : _ref$classNameMerchiC8,
    _ref$classNameMerchiC9 = _ref.classNameMerchiCheckoutButtonDownloadInvoice,
    classNameMerchiCheckoutButtonDownloadInvoice = _ref$classNameMerchiC9 === void 0 ? 'btn btn-lg btn-primary' : _ref$classNameMerchiC9,
    _ref$classNameMerchiC10 = _ref.classNameMerchiCheckoutFormCheckbox,
    classNameMerchiCheckoutFormCheckbox = _ref$classNameMerchiC10 === void 0 ? 'form-check-input' : _ref$classNameMerchiC10,
    _ref$classNameMerchiC11 = _ref.classNameMerchiCheckoutFormGroup,
    classNameMerchiCheckoutFormGroup = _ref$classNameMerchiC11 === void 0 ? 'form-group' : _ref$classNameMerchiC11,
    _ref$classNameMerchiC12 = _ref.classNameMerchiCheckoutFormGroupCheckbox,
    classNameMerchiCheckoutFormGroupCheckbox = _ref$classNameMerchiC12 === void 0 ? 'form-check' : _ref$classNameMerchiC12,
    _ref$classNameMerchiC13 = _ref.classNameMerchiCheckoutFormInput,
    classNameMerchiCheckoutFormInput = _ref$classNameMerchiC13 === void 0 ? 'form-control' : _ref$classNameMerchiC13,
    _ref$classNameMerchiC14 = _ref.classNameMerchiCheckoutGoogleSuggestList,
    classNameMerchiCheckoutGoogleSuggestList = _ref$classNameMerchiC14 === void 0 ? 'list-group m-b-0' : _ref$classNameMerchiC14,
    _ref$classNameMerchiC15 = _ref.classNameMerchiCheckoutGoogleSuggestListItem,
    classNameMerchiCheckoutGoogleSuggestListItem = _ref$classNameMerchiC15 === void 0 ? 'list-group-item cursor-pointer' : _ref$classNameMerchiC15,
    _ref$classNameMerchiC16 = _ref.classNameMerchiCheckoutListGroupItemLoader,
    classNameMerchiCheckoutListGroupItemLoader = _ref$classNameMerchiC16 === void 0 ? 'list-group-item modal-merchi-checkout-shipment-option' : _ref$classNameMerchiC16,
    _ref$classNameMerchiC17 = _ref.classNameMerchiCheckoutInputError,
    classNameMerchiCheckoutInputError = _ref$classNameMerchiC17 === void 0 ? 'text-danger' : _ref$classNameMerchiC17,
    _ref$classNameMerchiC18 = _ref.classNameMerchiCheckoutFormLabelCheckbox,
    classNameMerchiCheckoutFormLabelCheckbox = _ref$classNameMerchiC18 === void 0 ? '' : _ref$classNameMerchiC18,
    _ref$classNameMerchiC19 = _ref.classNameMerchiCheckoutListGroup,
    classNameMerchiCheckoutListGroup = _ref$classNameMerchiC19 === void 0 ? 'modal-merchi-checkout-shipment-option' : _ref$classNameMerchiC19,
    _ref$classNameMerchiC20 = _ref.classNameMerchiCheckoutListGroupItem,
    classNameMerchiCheckoutListGroupItem = _ref$classNameMerchiC20 === void 0 ? 'list-group-item' : _ref$classNameMerchiC20,
    _ref$classNameMerchiC21 = _ref.classNameMerchiCheckoutRow,
    classNameMerchiCheckoutRow = _ref$classNameMerchiC21 === void 0 ? 'merchi-row' : _ref$classNameMerchiC21,
    _ref$classNameMerchiC22 = _ref.classNameMerchiCheckoutRowColumn,
    classNameMerchiCheckoutRowColumn = _ref$classNameMerchiC22 === void 0 ? 'merchi-column' : _ref$classNameMerchiC22,
    _ref$classNameMerchiC23 = _ref.classNameMerchiCheckoutTabsContainer,
    classNameMerchiCheckoutTabsContainer = _ref$classNameMerchiC23 === void 0 ? 'merchi-checkout-tabs-container' : _ref$classNameMerchiC23,
    _ref$classNameMerchiC24 = _ref.classNameMerchiCheckoutTab,
    classNameMerchiCheckoutTab = _ref$classNameMerchiC24 === void 0 ? 'merchi-checkout-tab' : _ref$classNameMerchiC24,
    _ref$classNameMerchiC25 = _ref.classNameMerchiCheckoutTabButton,
    classNameMerchiCheckoutTabButton = _ref$classNameMerchiC25 === void 0 ? 'btn merchi-checkout-tab-btn' : _ref$classNameMerchiC25,
    _ref$classNameMerchiC26 = _ref.classNameMerchiCheckoutTabPane,
    classNameMerchiCheckoutTabPane = _ref$classNameMerchiC26 === void 0 ? 'tab-pane' : _ref$classNameMerchiC26,
    _ref$classNameMerchiC27 = _ref.classNameMerchiCheckoutTabPaneContainer,
    classNameMerchiCheckoutTabPaneContainer = _ref$classNameMerchiC27 === void 0 ? 'tab-content' : _ref$classNameMerchiC27,
    _ref$classNameMerchiI = _ref.classNameMerchiInvoiceButtonPayInvoice,
    classNameMerchiInvoiceButtonPayInvoice = _ref$classNameMerchiI === void 0 ? 'btn btn-lg btn-primary btn-block' : _ref$classNameMerchiI,
    currentUser = _ref.currentUser,
    _ref$discountButtonTe = _ref.discountButtonText,
    discountButtonText = _ref$discountButtonTe === void 0 ? 'Apply' : _ref$discountButtonTe,
    _ref$discountClassNam = _ref.discountClassName,
    discountClassName = _ref$discountClassNam === void 0 ? 'merchi-checkout-discount-code-container' : _ref$discountClassNam,
    _ref$discountClassNam2 = _ref.discountClassNameButton,
    discountClassNameButton = _ref$discountClassNam2 === void 0 ? 'btn btn-primary' : _ref$discountClassNam2,
    _ref$discountClassNam3 = _ref.discountClassNameButtonContainer,
    discountClassNameButtonContainer = _ref$discountClassNam3 === void 0 ? '' : _ref$discountClassNam3,
    _ref$discountClassNam4 = _ref.discountClassNameButtonItemRemove,
    discountClassNameButtonItemRemove = _ref$discountClassNam4 === void 0 ? 'btn btn-sm btn-link' : _ref$discountClassNam4,
    _ref$discountClassNam5 = _ref.discountClassNameErrorMessage,
    discountClassNameErrorMessage = _ref$discountClassNam5 === void 0 ? 'text-danger' : _ref$discountClassNam5,
    _ref$discountClassNam6 = _ref.discountClassNameInput,
    discountClassNameInput = _ref$discountClassNam6 === void 0 ? 'form-control' : _ref$discountClassNam6,
    _ref$discountClassNam7 = _ref.discountClassNameInputContainer,
    discountClassNameInputContainer = _ref$discountClassNam7 === void 0 ? 'merchi-checkout-discount-code-field-container' : _ref$discountClassNam7,
    _ref$discountClassNam8 = _ref.discountClassNameInputdiscountLabel,
    discountClassNameInputdiscountLabel = _ref$discountClassNam8 === void 0 ? 'visually-hidden' : _ref$discountClassNam8,
    _ref$discountClassNam9 = _ref.discountClassNameListItem,
    discountClassNameListItem = _ref$discountClassNam9 === void 0 ? 'list-group-item d-flex align-items-center justify-content-between mt-2' : _ref$discountClassNam9,
    _ref$discountClassNam10 = _ref.discountClassNameListItems,
    discountClassNameListItems = _ref$discountClassNam10 === void 0 ? 'list-group' : _ref$discountClassNam10,
    discountClassNameMainContainer = _ref.discountClassNameMainContainer,
    _ref$discountLabel = _ref.discountLabel,
    discountLabel = _ref$discountLabel === void 0 ? 'Discount code' : _ref$discountLabel,
    _ref$discountShowAppl = _ref.discountShowAppliedItems,
    discountShowAppliedItems = _ref$discountShowAppl === void 0 ? true : _ref$discountShowAppl,
    googlePlacesApiKey = _ref.googlePlacesApiKey,
    _ref$includeDomainSig = _ref.includeDomainSignup,
    includeDomainSignup = _ref$includeDomainSig === void 0 ? false : _ref$includeDomainSig,
    invoice = _ref.invoice,
    isBuyRequest = _ref.isBuyRequest,
    isOpen = _ref.isOpen,
    _ref$isProductEmbedFo = _ref.isProductEmbedForm,
    isProductEmbedForm = _ref$isProductEmbedFo === void 0 ? false : _ref$isProductEmbedFo,
    job = _ref.job,
    product = _ref.product,
    messageSuccessBuyRequest = _ref.messageSuccessBuyRequest,
    messageSuccessQuoteRequest = _ref.messageSuccessQuoteRequest,
    redirectAfterSuccessUrl = _ref.redirectAfterSuccessUrl,
    redirectAfterQuoteSuccessUrl = _ref.redirectAfterQuoteSuccessUrl,
    redirectWithValue = _ref.redirectWithValue,
    setInvoice = _ref.setInvoice,
    setJob = _ref.setJob,
    _ref$showDiscountCode = _ref.showDiscountCode,
    showDiscountCode = _ref$showDiscountCode === void 0 ? true : _ref$showDiscountCode,
    _ref$showUserTermsAnd = _ref.showUserTermsAndConditions,
    showUserTermsAndConditions = _ref$showUserTermsAnd === void 0 ? true : _ref$showUserTermsAnd,
    toggleMerchiCheckout = _ref.toggleMerchiCheckout,
    _ref$urlApi = _ref.urlApi,
    urlApi = _ref$urlApi === void 0 ? 'https://api.merchi.co/v6/' : _ref$urlApi,
    _ref$urlFrontend = _ref.urlFrontend,
    urlFrontend = _ref$urlFrontend === void 0 ? 'https://merchi.co/' : _ref$urlFrontend;
  var _useState = useState(0),
    _useState2 = _slicedToArray(_useState, 2),
    activeTabIndex = _useState2[0],
    setActiveTabIndex = _useState2[1];
  var _useState3 = useState(false),
    _useState4 = _slicedToArray(_useState3, 2),
    loading = _useState4[0],
    setLoading = _useState4[1];
  var _useState5 = useState([]),
    _useState6 = _slicedToArray(_useState5, 2),
    alerts = _useState6[0],
    setAlerts = _useState6[1];
  var _useState7 = useState(null),
    _useState8 = _slicedToArray(_useState7, 2),
    domain = _useState8[0],
    setDomain = _useState8[1];
  var _useState9 = useState([]),
    _useState10 = _slicedToArray(_useState9, 2),
    tabs = _useState10[0],
    setTabs = _useState10[1];
  var _useState11 = useState(false),
    _useState12 = _slicedToArray(_useState11, 2),
    googlePlacesLoaded = _useState12[0],
    setGoogleMapsLoaded = _useState12[1];
  function editDraftTemplate(index, draft) {
    var jobJson = _objectSpread({}, job);
    var draftTemplates = product.draftTemplates || [];
    var draftTemplateFiles = draftTemplates.map(function (d) {
      return d.file;
    });
    var ownDraft = jobJson.ownDrafts && jobJson.ownDrafts[0] || {
      images: draftTemplateFiles
    };
    if (ownDraft.images) {
      ownDraft.images[index] = draft;
    } else {
      ownDraft.images = [draft];
    }
    jobJson.ownDrafts = [ownDraft];
    setJob(jobJson);
  }
  function alertClose(index) {
    var updatedAlerts = alerts.filter(function (_, idx) {
      return idx !== index;
    });
    setAlerts(updatedAlerts);
  }
  function alertErrorShow(message) {
    var newAlerts = [].concat(_toConsumableArray(alerts), [{
      alertType: 'error',
      message: message,
      title: 'Error!'
    }]);
    setAlerts(newAlerts);
  }
  function resetTabs() {
    var _tabs = tabsInit({
      job: job,
      includeDomainSignup: includeDomainSignup,
      isBuyRequest: isBuyRequest
    });
    setTabs(_tabs);
  }
  function setCustomer(customer) {
    setJob(_objectSpread(_objectSpread({}, job), {}, {
      client: customer
    }));
    resetTabs();
  }
  function clearCustomer() {
    setCustomer({});
    resetTabs();
  }
  function nextTab() {
    var ndextIndex = activeTabIndex + 1;
    var _tabs = _toConsumableArray(tabs);
    _tabs[ndextIndex].disabled = false;
    setTabs(_tabs);
    setActiveTabIndex(ndextIndex);
  }
  function setActiveTabById(tabId) {
    var index = tabs.findIndex(function (t) {
      return t.id === tabId;
    });
    setActiveTabIndex(index);
  }
  // Persistent callback setup
  if (typeof window !== 'undefined') {
    if (!window.googleMapsScriptLoaded) {
      window.googleMapsScriptLoaded = function () {
        setGoogleMapsLoaded(true);
      };
    }
  }
  useEffect(function () {
    if (job.product && !tabs.length) {
      resetTabs();
    }
    if (!job.client) {
      setCustomer(currentUser || {});
    }
  }, [job]);
  useEffect(function () {
    // Use an ID for easier detection
    if (googlePlacesApiKey) {
      var scriptId = 'googleMapsScript';

      // Check if the script already exists
      var existingScript = document.getElementById(scriptId);
      // Check if any script from maps.googleapis.com already exists in the document
      var existingScript2 = document.querySelector("script[src^=\"https://maps.googleapis.com/maps-api-v3/api/js/\"]");
      if (!existingScript && !existingScript2) {
        var script = document.createElement('script');
        script.id = scriptId;
        script.src = "https://maps.googleapis.com/maps/api/js?key=".concat(googlePlacesApiKey, "&libraries=places&callback=googleMapsScriptLoaded");
        script.async = true;
        script.defer = true;
        // Append the script to the head
        document.head.appendChild(script);
      } else {
        setGoogleMapsLoaded(true);
      }
    }
  }, [googlePlacesApiKey]);
  function discountCallbackSuccess(items) {
    setJob(_objectSpread(_objectSpread({}, job), {}, {
      items: items
    }));
  }
  function discountCallbackError() {
    setJob(_objectSpread(_objectSpread({}, job), {}, {
      items: []
    }));
  }
  return /*#__PURE__*/_jsx(MerchiCheckoutContext.Provider, {
    value: {
      activeTabIndex: activeTabIndex,
      alerts: alerts,
      alertClose: alertClose,
      alertErrorShow: alertErrorShow,
      classNameMerchiCheckoutAlertsContainer: classNameMerchiCheckoutAlertsContainer,
      classNameMerchiCheckoutAlert: classNameMerchiCheckoutAlert,
      classNameMerchiCheckoutAlertError: classNameMerchiCheckoutAlertError,
      classNameMerchiCheckoutAlertSuccess: classNameMerchiCheckoutAlertSuccess,
      classNameMerchiCheckoutButtonPrimary: classNameMerchiCheckoutButtonPrimary,
      classNameMerchiCheckoutButtonPrimaryBlock: classNameMerchiCheckoutButtonPrimaryBlock,
      classNameMerchiCheckoutButtonSecondary: classNameMerchiCheckoutButtonSecondary,
      classNameMerchiCheckoutButtonSecondaryBlock: classNameMerchiCheckoutButtonSecondaryBlock,
      classNameMerchiCheckoutButtonDownloadInvoice: classNameMerchiCheckoutButtonDownloadInvoice,
      classNameMerchiCheckoutFormCheckbox: classNameMerchiCheckoutFormCheckbox,
      classNameMerchiCheckoutFormGroup: classNameMerchiCheckoutFormGroup,
      classNameMerchiCheckoutFormGroupCheckbox: classNameMerchiCheckoutFormGroupCheckbox,
      classNameMerchiCheckoutFormInput: classNameMerchiCheckoutFormInput,
      classNameMerchiCheckoutGoogleSuggestList: classNameMerchiCheckoutGoogleSuggestList,
      classNameMerchiCheckoutGoogleSuggestListItem: classNameMerchiCheckoutGoogleSuggestListItem,
      classNameMerchiCheckoutListGroupItemLoader: classNameMerchiCheckoutListGroupItemLoader,
      classNameMerchiCheckoutFormLabelCheckbox: classNameMerchiCheckoutFormLabelCheckbox,
      classNameMerchiCheckoutInputError: classNameMerchiCheckoutInputError,
      classNameMerchiCheckoutListGroup: classNameMerchiCheckoutListGroup,
      classNameMerchiCheckoutListGroupItem: classNameMerchiCheckoutListGroupItem,
      classNameMerchiCheckoutRow: classNameMerchiCheckoutRow,
      classNameMerchiCheckoutRowColumn: classNameMerchiCheckoutRowColumn,
      classNameMerchiCheckoutTabsContainer: classNameMerchiCheckoutTabsContainer,
      classNameMerchiCheckoutTab: classNameMerchiCheckoutTab,
      classNameMerchiCheckoutTabButton: classNameMerchiCheckoutTabButton,
      classNameMerchiCheckoutTabPane: classNameMerchiCheckoutTabPane,
      classNameMerchiCheckoutTabPaneContainer: classNameMerchiCheckoutTabPaneContainer,
      classNameMerchiInvoiceButtonPayInvoice: classNameMerchiInvoiceButtonPayInvoice,
      clearCustomer: clearCustomer,
      currentUser: currentUser,
      customer: job.client || {},
      discountButtonText: discountButtonText,
      discountCallbackError: discountCallbackError,
      discountCallbackSuccess: discountCallbackSuccess,
      discountClassName: discountClassName,
      discountClassNameButton: discountClassNameButton,
      discountClassNameButtonContainer: discountClassNameButtonContainer,
      discountClassNameButtonItemRemove: discountClassNameButtonItemRemove,
      discountClassNameErrorMessage: discountClassNameErrorMessage,
      discountClassNameInput: discountClassNameInput,
      discountClassNameInputContainer: discountClassNameInputContainer,
      discountClassNameInputdiscountLabel: discountClassNameInputdiscountLabel,
      discountClassNameListItem: discountClassNameListItem,
      discountClassNameListItems: discountClassNameListItems,
      discountClassNameMainContainer: discountClassNameMainContainer,
      discountLabel: discountLabel,
      discountShowAppliedItems: discountShowAppliedItems,
      domain: domain,
      editDraftTemplate: editDraftTemplate,
      googlePlacesLoaded: googlePlacesLoaded,
      includeDomainSignup: includeDomainSignup,
      invoice: invoice,
      isBuyRequest: isBuyRequest,
      isOpen: isOpen,
      isProductEmbedForm: isProductEmbedForm,
      job: job,
      loading: loading,
      messageSuccessBuyRequest: messageSuccessBuyRequest,
      messageSuccessQuoteRequest: messageSuccessQuoteRequest,
      nextTab: nextTab,
      product: product,
      redirectAfterSuccessUrl: redirectAfterSuccessUrl,
      redirectAfterQuoteSuccessUrl: redirectAfterQuoteSuccessUrl,
      redirectWithValue: redirectWithValue,
      setActiveTabById: setActiveTabById,
      setActiveTabIndex: setActiveTabIndex,
      setCustomer: setCustomer,
      setDomain: setDomain,
      setInvoice: setInvoice,
      setJob: setJob,
      showDiscountCode: showDiscountCode,
      showUserTermsAndConditions: showUserTermsAndConditions,
      tabs: tabs,
      toggleMerchiCheckout: toggleMerchiCheckout,
      urlApi: urlApi,
      urlFrontend: urlFrontend
    },
    children: children
  });
};
