// merchi_checkout/lib/components/MerchiCheckoutTabs.js
import * as React from 'react';
import { useMerchiCheckboutContext } from './MerchiCheckoutProvider';
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
function Tab(_ref) {
  var index = _ref.index,
    tab = _ref.tab;
  var _useMerchiCheckboutCo = useMerchiCheckboutContext(),
    activeTabIndex = _useMerchiCheckboutCo.activeTabIndex,
    classNameMerchiCheckoutTab = _useMerchiCheckboutCo.classNameMerchiCheckoutTab,
    classNameMerchiCheckoutTabButton = _useMerchiCheckboutCo.classNameMerchiCheckoutTabButton,
    setActiveTabIndex = _useMerchiCheckboutCo.setActiveTabIndex;
  var disabled = tab.disabled,
    title = tab.title;
  var classActive = activeTabIndex === index ? 'active' : '';
  return /*#__PURE__*/_jsxs("div", {
    className: classNameMerchiCheckoutTab,
    children: [/*#__PURE__*/_jsx("button", {
      disabled: disabled,
      className: "".concat(classNameMerchiCheckoutTabButton, " ").concat(classActive),
      style: {
        cursor: disabled ? 'not-allowed' : 'pointer'
      },
      onClick: function onClick() {
        if (!disabled) {
          setActiveTabIndex(index);
        }
      },
      children: index + 1
    }), /*#__PURE__*/_jsx("small", {
      className: "text-muted",
      children: title
    })]
  });
}
export default function MerchiCheckoutTabs() {
  var _useMerchiCheckboutCo2 = useMerchiCheckboutContext(),
    classNameMerchiCheckoutTabsContainer = _useMerchiCheckboutCo2.classNameMerchiCheckoutTabsContainer,
    tabs = _useMerchiCheckboutCo2.tabs;
  return /*#__PURE__*/_jsx("div", {
    className: classNameMerchiCheckoutTabsContainer,
    children: tabs.map(function (t, i) {
      return /*#__PURE__*/_jsx(Tab, {
        index: i,
        tab: t
      }, "".concat(i, "-tab-id"));
    })
  });
}
