// merchi_checkout/lib/components/TabPaneNewDomain.js
'use client';

import Cookies from 'js-cookie';
import { useMerchiCheckboutContext } from './MerchiCheckoutProvider';
import { tabIdComplete } from '../tabs_utils';
import FooterButtons from './FooterButtons';
import TitleStep from './TitleStep';
import TabPane from './TabPane';
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
function TabPaneNewDomain() {
  var _useMerchiCheckboutCo = useMerchiCheckboutContext(),
    domain = _useMerchiCheckboutCo.domain,
    setDomain = _useMerchiCheckboutCo.setDomain;
  var activeSession = Cookies.get('session_token');
  return /*#__PURE__*/_jsxs(TabPane, {
    tabId: tabIdComplete,
    children: [/*#__PURE__*/_jsx(TitleStep, {
      title: "Store - Domain Name"
    }), /*#__PURE__*/_jsxs("div", {
      className: "w-100",
      children: [/*#__PURE__*/_jsx("div", {
        className: "my-4",
        children: domain ? /*#__PURE__*/_jsx("p", {
          children: "Congratulations! ".concat(domain.domain, " has been successfuly created and is ready for you to start selling!")
        }) : /*#__PURE__*/_jsxs("p", {
          children: ["Create your free ", /*#__PURE__*/_jsx("a", {
            href: "/pricing",
            children: "Merchi Seller Store"
          }), " where you can sell your customised products once ", "they're", " ready."]
        })
      }), /*#__PURE__*/_jsx("div", {
        className: "create-new-domain-form-container"
      })]
    }), /*#__PURE__*/_jsx(FooterButtons, {})]
  });
}
export default TabPaneNewDomain;
