// merchi_checkout/lib/components/TabPaneCustomer.js
'use client';

import TabPane from './TabPane';
import TitleStep from './TitleStep';
import FooterButtons from './FooterButtons';
import { tabIdCustomerInfo } from '../tabs_utils';
import { isUserRegistered, isRegisteredAndHasStore } from '../utils';
import { useMerchiCheckboutContext } from './MerchiCheckoutProvider';
import { FormCustomerNew, FormCustomerReturning } from './forms';
import FormDomainNew from './forms/FormDomainNew';
import { BiUserCircle } from 'react-icons/bi';
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
export function CustomerInfo() {
  var _useMerchiCheckboutCo = useMerchiCheckboutContext(),
    job = _useMerchiCheckboutCo.job;
  var client = job.client;
  var _client$emailAddresse = client.emailAddresses,
    emailAddresses = _client$emailAddresse === void 0 ? [] : _client$emailAddresse,
    _client$name = client.name,
    name = _client$name === void 0 ? 'Returning customer' : _client$name,
    _client$phoneNumbers = client.phoneNumbers,
    phoneNumbers = _client$phoneNumbers === void 0 ? [] : _client$phoneNumbers;
  var emailAddress = emailAddresses[0] && emailAddresses[0].emailAddress || 'Email not shown';
  var phoneNumber = phoneNumbers[0] && phoneNumbers[0].internationalFormatNumber || 'Phone number not shown';
  return /*#__PURE__*/_jsxs("div", {
    className: "modal-merchi-checkout-customer-details text-left",
    children: [/*#__PURE__*/_jsx("strong", {
      children: name
    }), /*#__PURE__*/_jsx("p", {
      children: emailAddress
    }), /*#__PURE__*/_jsx("p", {
      children: phoneNumber
    })]
  });
}
export function SmallCustomerInfo() {
  var _useMerchiCheckboutCo2 = useMerchiCheckboutContext(),
    classNameMerchiCheckoutButtonPrimary = _useMerchiCheckboutCo2.classNameMerchiCheckoutButtonPrimary,
    clearCustomer = _useMerchiCheckboutCo2.clearCustomer;
  return /*#__PURE__*/_jsxs("div", {
    className: "w-100 modal-merchi-checkout-customer-card",
    children: [/*#__PURE__*/_jsx(CustomerInfo, {}), /*#__PURE__*/_jsx("div", {
      className: "mt-2",
      children: /*#__PURE__*/_jsxs("button", {
        className: classNameMerchiCheckoutButtonPrimary,
        onClick: clearCustomer,
        children: [/*#__PURE__*/_jsx(BiUserCircle, {
          fontSize: "1rem",
          style: {
            marginRight: '0.25rem'
          }
        }), ' ', "Change"]
      })
    })]
  });
}
function CustomerPanel() {
  var _useMerchiCheckboutCo3 = useMerchiCheckboutContext(),
    job = _useMerchiCheckboutCo3.job;
  var client = job.client;
  var isActive = isUserRegistered(client);
  return /*#__PURE__*/_jsx("div", {
    className: "".concat(isActive ? '' : 'hide'),
    children: client && /*#__PURE__*/_jsx(SmallCustomerInfo, {})
  });
}
function TabPaneCustomer() {
  var _useMerchiCheckboutCo4 = useMerchiCheckboutContext(),
    job = _useMerchiCheckboutCo4.job,
    includeDomainSignup = _useMerchiCheckboutCo4.includeDomainSignup;
  var client = job.client;
  var isActive = isUserRegistered(client);
  return /*#__PURE__*/_jsxs(TabPane, {
    tabId: tabIdCustomerInfo,
    children: [/*#__PURE__*/_jsx(TitleStep, {
      title: "Customer - Contact info"
    }), isActive ? /*#__PURE__*/_jsx(CustomerPanel, {}) : /*#__PURE__*/_jsx("div", {
      className: "".concat(isActive ? 'hide' : '', " d-flex justify-content-center flex-column"),
      children: /*#__PURE__*/_jsxs("div", {
        className: "w-100 customer-detail-form",
        children: [/*#__PURE__*/_jsxs("div", {
          className: "mt-1 mb-5",
          children: [/*#__PURE__*/_jsx("h5", {
            className: "mb-3",
            children: "Returning customer"
          }), /*#__PURE__*/_jsx(FormCustomerReturning, {})]
        }), /*#__PURE__*/_jsxs("div", {
          children: [/*#__PURE__*/_jsx("h5", {
            className: "mb-3",
            children: "New customer"
          }), /*#__PURE__*/_jsx(FormCustomerNew, {})]
        })]
      })
    }), client && client.id && !client.hasStore && includeDomainSignup && /*#__PURE__*/_jsx(FormDomainNew, {}), /*#__PURE__*/_jsx(FooterButtons, {
      forceDisabled: !isRegisteredAndHasStore(client),
      isActive: isActive
    })]
  });
}
export default TabPaneCustomer;
