// merchi_checkout/lib/components/TabPanePayment.js
'use client';

import { useMerchiCheckboutContext } from './MerchiCheckoutProvider';
import { tabIdPayment } from '../tabs_utils';
import TitleStep from './TitleStep';
import TabPane from './TabPane';
import MerchiInvoice from 'merchi_invoice';
import { redirectOnSuccess } from '../utils';
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
function TabPanePayment() {
  var _useMerchiCheckboutCo = useMerchiCheckboutContext(),
    alertErrorShow = _useMerchiCheckboutCo.alertErrorShow,
    classNameMerchiCheckoutButtonDownloadInvoice = _useMerchiCheckboutCo.classNameMerchiCheckoutButtonDownloadInvoice,
    classNameMerchiInvoiceButtonPayInvoice = _useMerchiCheckboutCo.classNameMerchiInvoiceButtonPayInvoice,
    invoice = _useMerchiCheckboutCo.invoice,
    nextTab = _useMerchiCheckboutCo.nextTab,
    redirectAfterSuccessUrl = _useMerchiCheckboutCo.redirectAfterSuccessUrl,
    redirectWithValue = _useMerchiCheckboutCo.redirectWithValue,
    setInvoice = _useMerchiCheckboutCo.setInvoice,
    urlApi = _useMerchiCheckboutCo.urlApi;
  return /*#__PURE__*/_jsxs(TabPane, {
    tabId: tabIdPayment,
    children: [/*#__PURE__*/_jsx(TitleStep, {
      title: "Checkout / Payment"
    }), /*#__PURE__*/_jsx("div", {
      className: "d-flex checkout-border-top",
      children: /*#__PURE__*/_jsx("div", {
        className: "w-100 mt-3",
        children: invoice && /*#__PURE__*/_jsx(MerchiInvoice, {
          alertErrorShow: alertErrorShow,
          callbackCreditCardPaymentSuccess: function callbackCreditCardPaymentSuccess(i) {
            if (redirectAfterSuccessUrl) {
              var _i$invoice;
              redirectOnSuccess(redirectAfterSuccessUrl, redirectWithValue, i === null || i === void 0 || (_i$invoice = i.invoice) === null || _i$invoice === void 0 ? void 0 : _i$invoice.totalCost);
            } else {
              setInvoice(i.invoice);
              nextTab();
            }
          },
          classNameMerchiInvoiceButtonDownloadInvoice: classNameMerchiCheckoutButtonDownloadInvoice,
          classNameMerchiInvoiceButtonPayInvoice: classNameMerchiInvoiceButtonPayInvoice,
          invoice: invoice,
          urlApi: urlApi
        })
      })
    })]
  });
}
export default TabPanePayment;
