// merchi_invoice/lib/stripe/StripePaymentButtons.js
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t["return"] && (u = t["return"](), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import { useEffect, useMemo, useState } from 'react';
import { PaymentRequestButtonElement, useStripe } from '@stripe/react-stripe-js';
import { jsx as _jsx } from "react/jsx-runtime";
function useOptions(paymentRequest) {
  var options = useMemo(function () {
    return {
      paymentRequest: paymentRequest,
      style: {
        paymentRequestButton: {
          theme: 'dark',
          height: '48px',
          type: 'buy'
        }
      }
    };
  }, [paymentRequest]);
  return options;
}
function usePaymentRequest(_ref) {
  var options = _ref.options,
    onPaymentMethod = _ref.onPaymentMethod;
  var stripe = useStripe();
  var _useState = useState(null),
    _useState2 = _slicedToArray(_useState, 2),
    paymentRequest = _useState2[0],
    setPaymentRequest = _useState2[1];
  var _useState3 = useState(false),
    _useState4 = _slicedToArray(_useState3, 2),
    canMakePayment = _useState4[0],
    setCanMakePayment = _useState4[1];
  useEffect(function () {
    if (stripe && paymentRequest === null) {
      var pr = stripe.paymentRequest(options);
      setPaymentRequest(pr);
    }
  }, [stripe, options, paymentRequest]);
  useEffect(function () {
    var subscribed = true;
    if (paymentRequest) {
      paymentRequest.canMakePayment().then(function (res) {
        if (res && subscribed) {
          setCanMakePayment(true);
        }
      });
    }
    return function () {
      subscribed = false;
    };
  }, [paymentRequest]);
  useEffect(function () {
    if (paymentRequest) {
      paymentRequest.on('paymentmethod', onPaymentMethod);
    }
    return function () {
      if (paymentRequest) {
        paymentRequest.off('paymentmethod', onPaymentMethod);
      }
    };
  }, [paymentRequest, onPaymentMethod]);
  return canMakePayment ? paymentRequest : null;
}
function StripePaymentButtons(_ref2) {
  var doPaymentRequest = _ref2.doPaymentRequest,
    invoice = _ref2.invoice;
  var paymentRequest = usePaymentRequest({
    options: {
      country: invoice.domain.company.country,
      currency: String(invoice.currency).toLowerCase(),
      total: {
        label: "Merchi Invoice #".concat(invoice.id),
        amount: Math.round(invoice.totalCost * 100)
      }
    },
    onPaymentMethod: doPaymentRequest
  });
  var options = useOptions(paymentRequest);
  if (!paymentRequest) {
    return null;
  }
  return /*#__PURE__*/_jsx("div", {
    style: {
      marginBottom: '2rem',
      marginTop: '2rem'
    },
    children: /*#__PURE__*/_jsx(PaymentRequestButtonElement, {
      className: "PaymentRequestButton",
      options: options
    })
  });
}
export default StripePaymentButtons;
