// merchi_invoice/lib/InvoiceInfo.js
'use client';

import ButtonInvoiceDownload from './buttons/ButtonInvoiceDownload';
import { jsxs as _jsxs } from "react/jsx-runtime";
import { jsx as _jsx } from "react/jsx-runtime";
import { Fragment as _Fragment } from "react/jsx-runtime";
function InvoiceInfo(_ref) {
  var alertErrorShow = _ref.alertErrorShow,
    classNameMerchiInvoiceButtonDownloadInvoice = _ref.classNameMerchiInvoiceButtonDownloadInvoice,
    invoice = _ref.invoice,
    urlApi = _ref.urlApi;
  var id = invoice.id,
    unpaid = invoice.unpaid;
  var paid = !unpaid;
  return /*#__PURE__*/_jsxs("div", {
    className: "w-100 d-flex flex-column align-items-center",
    children: [/*#__PURE__*/_jsxs("h5", {
      className: "container-invoice-heading mb-0",
      children: ["Invoice #", id]
    }), paid && /*#__PURE__*/_jsx(_Fragment, {
      children: /*#__PURE__*/_jsx("p", {
        className: "mt-0 mb-0",
        children: "Paid, thank you!"
      })
    }), /*#__PURE__*/_jsxs("p", {
      className: "mt-2 mb-2",
      children: ["Download your ", paid ? 'receipt' : 'invoice', "."]
    }), /*#__PURE__*/_jsx("div", {
      className: "d-block mt-1 mb-5",
      children: /*#__PURE__*/_jsx(ButtonInvoiceDownload, {
        alertErrorShow: alertErrorShow,
        classNameMerchiInvoiceButtonDownloadInvoice: classNameMerchiInvoiceButtonDownloadInvoice,
        invoice: invoice,
        urlApi: urlApi
      })
    })]
  });
}
export default InvoiceInfo;
