// merchi_checkout/lib/components/lists/ListShipmentQuoteOptions.js
'use client';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCheck } from '@fortawesome/free-solid-svg-icons';
import { CgSpinner } from 'react-icons/cg';
import { useMerchiCheckboutContext } from '../MerchiCheckoutProvider';
import { addressInOneLine } from '../../utils';
import { currencyTaxAndCost } from '../currency';
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
import { Fragment as _Fragment } from "react/jsx-runtime";
function ShipmentPrice(_ref) {
  var shipment = _ref.shipment;
  var shipmentMethod = shipment.shipmentMethod,
    taxType = shipment.taxType;
  var currency = shipmentMethod.currency;
  return /*#__PURE__*/_jsx("div", {
    children: !!shipment.cost && /*#__PURE__*/_jsx("small", {
      className: "font-weight-bold",
      children: currencyTaxAndCost(currency, taxType, shipment.cost)
    })
  });
}
function PickupInfo(_ref2) {
  var originAddress = _ref2.originAddress;
  return /*#__PURE__*/_jsxs("small", {
    children: ["Pick up from: ", addressInOneLine(originAddress)]
  });
}
function ShipmentOptionInfo(_ref3) {
  var shipment = _ref3.shipment;
  var name = shipment.name,
    shipmentMethod = shipment.shipmentMethod,
    transportCompanyName = shipment.transportCompanyName;
  var originAddress = shipmentMethod.originAddress,
    pickUp = shipmentMethod.pickUp;
  return /*#__PURE__*/_jsxs("div", {
    className: "shipment-option-info",
    children: [!name ? /*#__PURE__*/_jsxs(_Fragment, {
      children: [!!transportCompanyName && /*#__PURE__*/_jsx("p", {
        className: "mb-0",
        children: shipmentMethod.name
      }), pickUp ? /*#__PURE__*/_jsx(PickupInfo, {
        originAddress: originAddress
      }) : /*#__PURE__*/_jsx("small", {
        children: transportCompanyName
      })]
    }) : /*#__PURE__*/_jsxs(_Fragment, {
      children: [/*#__PURE__*/_jsx("p", {
        className: "m-0",
        children: name
      }), pickUp && /*#__PURE__*/_jsx(PickupInfo, {
        originAddress: originAddress
      })]
    }), /*#__PURE__*/_jsx(ShipmentPrice, {
      shipment: shipment
    })]
  });
}
function ListItemShipmentQuoteOption(_ref4) {
  var doSelect = _ref4.doSelect,
    isSelected = _ref4.isSelected,
    shipment = _ref4.shipment;
  var _useMerchiCheckboutCo = useMerchiCheckboutContext(),
    classNameMerchiCheckoutListGroupItem = _useMerchiCheckboutCo.classNameMerchiCheckoutListGroupItem;
  return /*#__PURE__*/_jsx("div", {
    className: "".concat(classNameMerchiCheckoutListGroupItem, " ").concat(isSelected ? 'active' : ''),
    style: {
      cursor: 'pointer'
    },
    onClick: doSelect,
    children: /*#__PURE__*/_jsxs("div", {
      style: {
        display: 'flex',
        justifyContent: 'space-between',
        alignItems: 'center'
      },
      children: [/*#__PURE__*/_jsx(ShipmentOptionInfo, {
        shipment: shipment
      }), /*#__PURE__*/_jsx("div", {
        children: isSelected && /*#__PURE__*/_jsx(FontAwesomeIcon, {
          color: "white",
          icon: faCheck
        })
      })]
    })
  });
}
function ListItemLoading() {
  var _useMerchiCheckboutCo2 = useMerchiCheckboutContext(),
    classNameMerchiCheckoutListGroupItemLoader = _useMerchiCheckboutCo2.classNameMerchiCheckoutListGroupItemLoader;
  return /*#__PURE__*/_jsx("div", {
    className: classNameMerchiCheckoutListGroupItemLoader + ' d-flex align-items-center',
    children: /*#__PURE__*/_jsx(CgSpinner, {
      fontSize: "1.5rem",
      className: "animate_spin"
    })
  });
}
function ListItemNoOptions() {
  var _useMerchiCheckboutCo3 = useMerchiCheckboutContext(),
    classNameMerchiCheckoutListGroupItem = _useMerchiCheckboutCo3.classNameMerchiCheckoutListGroupItem;
  return /*#__PURE__*/_jsxs("div", {
    className: classNameMerchiCheckoutListGroupItem,
    children: [/*#__PURE__*/_jsx("div", {
      className: "merchi_monster_shipment_option_not_found"
    }), /*#__PURE__*/_jsx("small", {
      className: "text-muted shipment-option-checkout-form",
      children: "Enter your address to select a shipment option"
    })]
  });
}
function ListShipmentQuoteOptions(_ref5) {
  var doSelectShipmentOption = _ref5.doSelectShipmentOption,
    loading = _ref5.loading,
    selectedOption = _ref5.selectedOption,
    shipmentOptions = _ref5.shipmentOptions;
  var _useMerchiCheckboutCo4 = useMerchiCheckboutContext(),
    classNameMerchiCheckoutListGroup = _useMerchiCheckboutCo4.classNameMerchiCheckoutListGroup;
  var index = selectedOption ? shipmentOptions.map(function (s) {
    return s.shipment;
  }).findIndex(function (item) {
    return item.id === selectedOption.id;
  }) : null;
  return /*#__PURE__*/_jsx("div", {
    className: classNameMerchiCheckoutListGroup,
    children: loading ? /*#__PURE__*/_jsx(ListItemLoading, {}) : shipmentOptions.length ? shipmentOptions.map(function (s, i) {
      return /*#__PURE__*/_jsx(ListItemShipmentQuoteOption, {
        doSelect: function doSelect() {
          return doSelectShipmentOption(s.shipment);
        },
        isSelected: index === i,
        shipment: s.shipment
      }, "".concat(i, "-option"));
    }) : /*#__PURE__*/_jsx(ListItemNoOptions, {})
  });
}
export default ListShipmentQuoteOptions;
