// merchi_checkout/lib/components/merchi_image_editor/EditorImages.js
import React from 'react';
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
import { Fragment as _Fragment } from "react/jsx-runtime";
var DraftTemplateThumbnail = function DraftTemplateThumbnail(_ref) {
  var draft = _ref.draft;
  // Implement the DraftTemplateThumbnail here.
  // I don't have the details so I'm just rendering a placeholder.

  return /*#__PURE__*/_jsx("img", {
    style: {
      objectFit: 'cover'
    },
    src: draft.file.viewUrl,
    height: "100",
    width: "100",
    alt: "Draft Thumbnail"
  });
};
var EditorImages = function EditorImages(_ref2) {
  var activeIndex = _ref2.activeIndex,
    _ref2$classNameEditor = _ref2.classNameEditorImagesContainer,
    classNameEditorImagesContainer = _ref2$classNameEditor === void 0 ? 'd-flex mb-3' : _ref2$classNameEditor,
    _ref2$classNameEditor2 = _ref2.classNameEditorImagesImage,
    classNameEditorImagesImage = _ref2$classNameEditor2 === void 0 ? 'nav-item mr-2' : _ref2$classNameEditor2,
    draftTemplates = _ref2.draftTemplates,
    setActiveTemplateIndex = _ref2.setActiveTemplateIndex;
  function isActive(index) {
    return index === activeIndex;
  }
  var activeDraftTemplate = draftTemplates[activeIndex];
  return /*#__PURE__*/_jsxs(_Fragment, {
    children: [/*#__PURE__*/_jsx("div", {
      className: classNameEditorImagesContainer,
      children: draftTemplates.map(function (d, index) {
        return /*#__PURE__*/_jsx("div", {
          className: "".concat(classNameEditorImagesImage, " ").concat(isActive(index) ? 'active' : ''),
          onClick: function onClick() {
            return setActiveTemplateIndex(index);
          },
          style: {
            borderRadius: '0.75rem',
            border: isActive(index) ? '2px #888 dashed' : '2px rgb(213, 213, 213) dashed',
            height: 101,
            overflow: 'hidden',
            width: 101,
            cursor: 'pointer',
            transition: 'border-color 0.2s ease-in-out'
          },
          children: /*#__PURE__*/_jsx(DraftTemplateThumbnail, {
            draft: d
          })
        }, "".concat(index, "-draft-image"));
      })
    }), activeDraftTemplate ? /*#__PURE__*/_jsxs("div", {
      className: "d-flex flex-column",
      children: [/*#__PURE__*/_jsxs("h4", {
        className: "mb-0",
        children: [activeDraftTemplate.name, " ", activeDraftTemplate.height, " *", ' ', activeDraftTemplate.width, "px"]
      }), /*#__PURE__*/_jsx("small", {
        className: "mt-0",
        children: activeDraftTemplate.description
      })]
    }) : /*#__PURE__*/_jsx("div", {
      children: /*#__PURE__*/_jsx("h5", {
        className: "mb-0",
        children: "Loading template..."
      })
    })]
  });
};
export default EditorImages;
