// merchi_checkout/lib/components/merchi_image_editor/MerchiImageEditor.js
import ImageEditor from './ImageEditor';
import EditorImages from './EditorImages';
import { jsx as _jsx } from "react/jsx-runtime";
import { Fragment as _Fragment } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
export default function MerchiImageEditor(_ref) {
  var activeTemplateIndex = _ref.activeTemplateIndex,
    clientFiles = _ref.clientFiles,
    draftTemplates = _ref.draftTemplates,
    _editDraftTemplate = _ref.editDraftTemplate,
    setActiveTemplateIndex = _ref.setActiveTemplateIndex,
    setClientFiles = _ref.setClientFiles;
  return /*#__PURE__*/_jsxs(_Fragment, {
    children: [/*#__PURE__*/_jsx(EditorImages, {
      activeIndex: activeTemplateIndex,
      draftTemplates: draftTemplates,
      setActiveTemplateIndex: setActiveTemplateIndex
    }), /*#__PURE__*/_jsx("div", {
      style: {
        border: '1px solid f0f0f0',
        padding: 10,
        margin: '10px 0',
        borderRadius: 10,
        boxShadow: '0px 5px 20px rgba(0,0,0,0.05)'
      },
      children: draftTemplates.map(function (t, index) {
        return /*#__PURE__*/_jsx(ImageEditor, {
          clientFiles: clientFiles,
          editDraftTemplate: function editDraftTemplate(merchiFile) {
            return _editDraftTemplate(index, merchiFile);
          },
          draftTemplate: t,
          isActive: activeTemplateIndex === index,
          setClientFiles: setClientFiles
        }, index);
      })
    })]
  });
}
